Faster random number generation in fused_rowwise_random_quantization_ops

wenwei202 opened the pull request:
Summary:
Depends on D8745940

```
Trying example: test_speed_of_rand_quantization(self=<caffe2.caffe2.fb.net_transforms.tests.rand_quantization_op_speed_test.TestSpeedFloatToFusedRandRowwiseQuantized testMethod=test_speed_of_rand_quantization>, bitwidth_=2, random_=True
, data_shape_=array([1024, 1224]), gc=, dc=[])
Sub+Scale+Sum time: 1.28631591796875 ms
Quantizing time: 1.3524532318115234 ms (1.0514160735468565X)
De-quantizing time: 0.43795108795166016 ms (0.34046930604982206X)
```

Differential Revision: D8849770

harouwu reviewed (approved):
Operator LG. Please wait until checks are passed before merging

Commit:
Faster random number generation in fused_rowwise_random_quantization_ops (#10634)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/10634

Depends on D9347706

```
Trying example: test_speed_of_rand_quantization(self=<caffe2.caffe2.python.operator_test.rand_quantization_op_speed_test.TestSpeedFloatToFusedRandRowwiseQuantized testMethod=test_speed_of_rand_quantization>, bitwidth_=2, random_=True, data_shape_=array([1024, 1224]), gc=, dc=[, device_type: 1])
Sub+Scale+Sum time: 1.9944190979003908 ms
Quantizing time: 2.080512046813965 ms (1.0431669296609765X)
De-quantizing time: 0.7375001907348633 ms (0.36978195380863577X)
```

```
Trying example: test_speed_of_rand_quantization(self=<caffe2.caffe2.python.operator_test.rand_quantization_op_speed_test.TestSpeedFloatToFusedRandRowwiseQuantized testMethod=test_speed_of_rand_quantization>, bitwidth_=1, random_=True, data_shape_=array([1024, 1224]), gc=device_type: 1, dc=[, device_type: 1])
Sub+Scale+Sum time: 1.6691923141479492 ms
Quantizing time: 7.500243186950684 ms (4.493336761366071X)
De-quantizing time: 1.1209726333618164 ms (0.6715658967876477X)
```

Differential Revision: D8849770

fbshipit-source-id: bf7f306bc085e3160ad5b7a0f81ae8ff823d2a3a
